src/main.cpp

```diff
@@ -603,10 +610,27 @@ static void write_safely(const string &name, const string &content) {
 	}
 }
 
+static string pt_str(const ptree &pt, const string& key) {
+	auto val = pt.get<std::string>(key);
+	val.erase(std::remove(val.begin(), val.end(), '\"'), val.end());
+	return val;
+}
+
+static void parse_os_release(OsRelease &osrelease) {
+	ptree pt;
+	boost::property_tree::ini_parser::read_ini("/etc/os-release", pt);
+	osrelease.factory = pt_str(pt, "LMP_FACTORY");
+	osrelease.hwid = pt_str(pt, "LMP_MACHINE");
+	osrelease.tag = pt_str(pt, "LMP_FACTORY_TAG");
```

Comment:
I often use `lmp-device-register` locally outside the LmP device during development.

I actually _think_ that device manufacturers might want to batch-register large swarms of devices the same way.

So:
- at the least, I think this `os-release` parser should feel OK when any of these values is not present.
- ideally, it would allow to build a factory-specific `lmp-device-register` _the old way_ if the user needs to.

Reply:
IIUC @mwasilew version in #34 has these values optional, just like I suggested. So, I'm fine.

Mind me, I'm a regular lazy developer who doesn't want to patch os-release every time I want to test lmp-device-register or API changes.

Reply:
I agree that lmp-dev-reg should work regardless of the `os-release` file presence.